src/ctap/mod.rs

```diff
@@ -3746,6 +3748,28 @@ mod test {
         assert_eq!(response, Err(Ctap2StatusCode::CTAP2_ERR_NOT_ALLOWED));
     }
 
+    #[test]
+    fn test_check_user_presence() {
+        // This TestEnv always returns successful user_presence checks.
+        let mut env = TestEnv::new();
+        let response = check_user_presence(&mut env, DUMMY_CHANNEL);
+        assert!(matches!(response, Ok(_)));
+    }
+
+    #[test]
+    fn test_check_user_presence_timeout() {
+        // This will always return timeout.
+        fn user_presence_timeout() -> UserPresenceResult {
+            Err(UserPresenceError::Timeout)
+        }
+
+        let mut env = TestEnv::new();
+        env.user_presence().set(user_presence_timeout);
+        let response = check_user_presence(&mut env, DUMMY_CHANNEL);
+        // TODO(https://github.com/google/OpenSK/issues/519): This should be timeout.
+        assert!(matches!(response, Ok(_)));
```

Comment:
Feel free to add this commit to #521, then we can have the correct assert here right away no?

Reply:
OK, done, added to #521.